src/helpers/agendaHelpers.ts

```diff
@@ -0,0 +1,209 @@
+import type { AgendaSession } from "./getAgenda";
+
+// Helper function for getting category color (server-side version)
+export function getTalkCategoryColor(session: AgendaSession): string {
+  const track = getTrackName(session);
+
+  if (!track) return "#506592";
+
+  const categoryName = track.toLowerCase();
+  const colorMap: Record<string, string> = {
+    frontend: "#fd6a82",
+    front: "#fd6a82",
+    backend: "#f78750",
+    back: "#f78750",
+    ai: "#81c47a",
+    "artificial intelligence": "#81c47a",
+    data: "#81c47a",
+    mobile: "#506592",
+    devops: "#9d4edd",
+    security: "#e63946",
+    "ui/ux": "#f72585",
+    design: "#f72585",
+    ignites: "#f7ab22",
+    opening: "#7ebec8",
+    keynote: "#7ebec8",
+    registration: "#5065926e",
+    dining: "transparent",
+  };
+
+  return colorMap[categoryName] || "#506592";
+}
+
+// Helper function to get track name from session
+export function getTrackName(session: AgendaSession): string | null {
+  if (session.isLightningGroup) return "Ignites";
+  if (session.title === "Opening Words") return "opening";
+  if (session.room === "Dining Hall") return "dining";
+  if (session.title.startsWith("Registration")) return "Registration";
+  if (session.title === "Keynote placeholder" || session.title === "Keynote Placeholder") return "keynote";
+  const trackCategory = session.categories?.find((c) => c.name === "Track");
+  const item = trackCategory?.categoryItems?.[0];
+  if (!item) return null;
+  const raw = item.name;
+  if (raw === "AI Apps" || raw === "AI Infra") return "AI";
+  return raw;
+}
+
+// Helper function to create URL-friendly slug
+export function createSlug(text: string): string {
+  return text
+    .toLowerCase()
+    .replace(/[^\w\s-]/g, "") // Remove special characters
+    .replace(/\s+/g, "-") // Replace spaces with hyphens
+    .replace(/-+/g, "-") // Replace multiple hyphens with single
+    .replace(/^-+|-+$/g, ""); // Remove leading/trailing hyphens
+}
+
+// Helper function to create speaker URL
+export function createSpeakerUrl(speakerName: string): string {
+  return `/speakers/${createSlug(speakerName)}`;
+}
+
+// Helper function to create talk URL
+export function createTalkUrl(talkTitle: string): string {
+  return `/talks/${createSlug(talkTitle)}`;
```

Comment:
likewise I think Singular is better 
Also, I personally prefer "session" over "talk" but not a biggie

Reply:
@rantav  I agree, I'll do it in a separate PR